app/src/main/java/com/example/androidtrainingproject/ui/home/CategoriesAccordion.kt

```diff
@@ -0,0 +1,126 @@
+import androidx.compose.material3.Card
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.rotate
+import androidx.compose.ui.graphics.Color.Companion.Transparent
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import com.example.androidtrainingproject.R
+import com.example.androidtrainingproject.ui.theme.Black
+import com.example.androidtrainingproject.ui.theme.Purple
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun CategoriesAccordion(
+    homeViewModel: HomeViewModel,
+    checkedCategories: List<String?>,
+    checkCategory: (String) -> Unit,
+    unCheckCategory: (String) -> Unit
+) {
+    var expanded by remember { mutableStateOf(false) }
+    val rotateState = animateFloatAsState(targetValue = if (expanded) 180f else 0f)
+
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(top = dimensionResource(id = R.dimen.padding_xl))
+    ) {
+        Card(
+            onClick = { expanded = !expanded },
+            colors = CardDefaults.cardColors(
+                containerColor = Transparent,
+                contentColor = Black
+            )
+        ) {
+            Row(
+                modifier = Modifier.fillMaxWidth(),
+                horizontalArrangement = Arrangement.SpaceBetween,
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                Text(
+                    text = stringResource(id = R.string.product_category_label),
+                    modifier = Modifier.fillMaxWidth(0.90f),
+                    style = MaterialTheme.typography.titleSmall
+                )
+                Icon(
+                    modifier = Modifier.rotate(rotateState.value),
+                    painter = painterResource(id = R.drawable.expand_more),
+                    contentDescription = null
+                )
+            }
+        }
+
+        AnimatedVisibility(visible = expanded) {
+            Column(modifier = Modifier.fillMaxWidth()) {
+                LazyColumn(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(horizontal = dimensionResource(id = R.dimen.padding_sm))
+                ) {
+                    items(homeViewModel.allCategories.size) { index ->
+                        val category = homeViewModel.allCategories[index]
```

Comment:
Always extract list items to their separate composables. This can be a `CategoriesAccordionItem` composable here in the same file below `CategoriesAccordion`.